### ISSUE TITLE
ERROR 1005 (HY000): Can't create table 'reporting.report' (errno: 150)

### ISSUE THREAD

**Issue by papillon326** [Icinga/icingaweb2-module-reporting#42]
## Describe the bug
According install procedure, "ERROR 1005 (HY000): Can't create table 'reporting.report' (errno: 150)" occured at importing "mysql.sql" scheme.

## To Reproduce
See above.

## Expected behavior
No error.

## Screenshots
N/A

## Your Environment
* Module version: 0.10.0 (dae92c9)
* Dependent module versions: 
* Icinga Web 2 version and modules (System - About): 2.7.3
* Web browser and version: N/A
* Icinga 2 version used (`icinga2 --version`): 2.11.2-1
* PHP version used (`php --version`): 7.2.24
* Server operating system and version: CentOS 7.7
* Database: MariaDB 5.5.64

## Additional context
This problem is caused by "Foreign key constraint". "template" table scheme is added in unsuitable location. Add "SET foreign_key_checks = 0;" to beginng of the file, I can avoid this problem.

**Comment by NeverUsedID** [comment]
Same here. Thanks for workaround.

**Comment by MarcusCaepio** [comment]
I can confirm this bug

**Comment by kyob** [comment]
Same here:
```sql
MariaDB [reporting]> source mysql.sql;
Query OK, 0 rows affected (0.039 sec)

Query OK, 13 rows affected (0.008 sec)
Records: 13  Duplicates: 0  Warnings: 0

ERROR 1005 (HY000) at line 28 in file: 'mysql.sql': Can't create table `reporting`.`report` (errno: 150 "Foreign key constraint is incorrectly formed")
ERROR 1005 (HY000) at line 42 in file: 'mysql.sql': Can't create table `reporting`.`reportlet` (errno: 150 "Foreign key constraint is incorrectly formed")
ERROR 1005 (HY000) at line 52 in file: 'mysql.sql': Can't create table `reporting`.`config` (errno: 150 "Foreign key constraint is incorrectly formed")
ERROR 1005 (HY000) at line 63 in file: 'mysql.sql': Can't create table `reporting`.`schedule` (errno: 150 "Foreign key constraint is incorrectly formed")
Query OK, 0 rows affected (0.031 sec)
```

**Comment by austinjhunt** [comment]
Alternatively (more robustly), you can move the CREATE TABLE template section above the CREATE TABLE report section since the report table contains an FK mapped to template.

**Comment by lippserd** [comment]
Fixed w/ #51
